src/ucp/api/ucp.h

```diff
@@ -333,6 +335,19 @@ typedef enum {
 
 /**
  * @ingroup UCP_DATATYPE
+ * @brief Generate a reusable identifier for Scatter-gather IOV data type.
```

Comment:
1. i suggest that the API will not have reusable flag at all, but the behavior will be at least as optimal as if is_reusable==1 in your suggestion
2. yes, a new type which we need in order to support MPI_Type_struct. The main difference from IOV is that it's **without buffer pointers**. The question is if the use case for persistent IOV can be satisfied with struct type.